src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,192 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.RegistrationException;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int AGE_18 = 18;
+    private static final int AGE_UNDER_18 = 10;
+    private static final int NEGATIVE_AGE = -1;
+    private static final int MAX_AGE = 105;
+    private static final int OVER_MAX_AGE = 106;
+    private static final int AGE_OVER_18 = 20;
+    private static final String USER_NAME = "Docent";
+    private static final String NEW_USER_NAME = "Daisy";
+    private static final String USER_PASSWORD_OK = "123456";
+    private static final String USER_PASSWORD_THREE_CHARS = "123";
+    private static final String USER_PASSWORD_FIVE_CHARS = "12345";
+    private static final String USER_PASSWORD_LENGTH_OK = "1234567";
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final long USER_ID = 123L;
+    private static RegistrationService registrationService;
+    private static User user;
+    private static User newUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin(USER_NAME);
+        user.setAge(AGE_18);
+        user.setPassword(USER_PASSWORD_OK);
+
+        newUser = new User();
+        newUser.setLogin(USER_NAME);
+        newUser.setAge(AGE_18);
+        newUser.setPassword(USER_PASSWORD_OK);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        user = null;
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+            throw new RegistrationException(" ");
+        });

```

Comment:
throw new RegistrationException В вередині асерту? А що це за тест? Ось даю приклад тесту ще, перевір чи працює)
```
        assertThrows(RegistrationException.class, () -> {
            int a = 2;
            throw new RegistrationException(" ");
        });
```

```suggestion
        assertThrows(RegistrationException.class, () -> {
            registrationService.register(null);
        });
```